flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -150,6 +160,29 @@ long getWindowTime() {
 		}
 
 		/**
+		 * Check pattern after match skip strategy.
+		 */
```

Comment:
Could this whole check go in the `Pattern` class? This will make the code clearer.